### PR TITLE
remove PodFile.lock

### PR DESCRIPTION
**Pull request by ccen-stripe**
So that when running `pod install` we always download the latest pods

### CODE REVIEW

**Comment by ruben-stripe** [comment]
I am not sure if it is recommended to `.gitignore` `Podfile.lock`. Each user will be using different versions of the same library and it may bring different errors for different users. @mludowise-stripe what is the recommended approach?

**Comment by ccen-stripe** [comment]
> I am not sure if it is recommended to `.gitignore` `Podfile.lock`. Each user will be using different versions of the same library and it may bring different errors for different users. @mludowise-stripe what is the recommended approach?

yeah Mel suggested this. The previous Podlock file uses `22.1.0`, which breaks the build. removing this would ensure we always get the latest one(currently `22.6.0`)

**Comment by ccen-stripe** [comment]
> > I am not sure if it is recommended to `.gitignore` `Podfile.lock`. Each user will be using different versions of the same library and it may bring different errors for different users. @mludowise-stripe what is the recommended approach?
> 
> A production project should include a `Podfile.lock` file, but the purpose of the Example app in this repo is to demo the SDK so removing it ensures that `pod install` will always pull the latest. But an unintended consequence could be that we're no longer locking the other pods (e.g. `Flipper`, `React-Core`, etc).
> 
> An alternative approach is to instead update the README's instructions for running the example app to include instructions to update.
> 
> ```
> * Install pods for iOS
>    * `cd ios && pod install`
> ```
> 
> to
> 
> ```
> * Install pods for iOS and update StripeIdentity
>    * `cd ios && pod install && pod update StripeIdentity`
> ```

Should we instead always manually update the lock file when there's a new `StripeIdentity` release? it seems the difference between `.lock` and `.podspec` here won't help much if we always need the latest identity release

**Comment by ccen-stripe** [comment]
> We could manually update the lock files and also follow @mludowise-stripe recommendation (update README) in case we forget to update the file after an update

SG! updated

File: README.md [hunk 1]
```diff
@@ -182,7 +182,7 @@ import type {
 - Run this command from root folder
   - `cd example && yarn`
 - Install pods for iOS
```
Comment (ruben-stripe) [inline README.md:184]: nit: `... and update StripeIdentity`

**Review by ruben-stripe — APPROVED** [review]

**Pull request merged**